src/Core/Startup/ErrorControlChainMiddleware.php

```diff
@@ -88,8 +94,19 @@ protected function safeReloadWithToken(HTTPRequest $request, $reloadToken)
         $request->getSession()->init($request);
 
         // Next, check if we're in dev mode, or the database doesn't have any security data, or we are admin
-        if (Director::isDev() || !Security::database_is_ready() || Permission::check('ADMIN')) {
-            return $reloadToken->reloadWithToken();
+        $reload = function (HTTPRequest $request) use ($reloadToken) {
```

Comment:
I think if any part changes, it won't be this. Using a callback as the 'action' is pretty standard for the middleware pattern.